### ISSUE TITLE
Incorrect Conv3d weights initialization?

### ISSUE THREAD

**Issue by pkdogcom** [kenshohara/3D-ResNets-PyTorch#23]
The Resnet and ResNeXt models (I haven't checked others) seem to be trying to initialize the weights using kaiming's initialization method in his Resnet paper using a normal distribution. However, by comparing the codes with the details in the paper as well as pytorch's own implementation (yes, pytorch has implemented kaiming's initializations), the Conv3d weight initialization seems to miss calculating the size of the third dimension in the kernel, i.e. instead of using 
```
if isinstance(m, nn.Conv3d):
    # Kernel is 3D but here only considers the time and row
    n = m.kernel_size[0] * m.kernel_size[1] * m.out_channels 
    m.weight.data.normal_(0, math.sqrt(2. / n))
```
we can use pytorch's implementation directly:
```
nn.init.kaiming_normal(m.weight, mode='fan_out')
```
whose fan in/out factor is calculated by 
```
        num_input_fmaps = tensor.size(1)
        num_output_fmaps = tensor.size(0)
        receptive_field_size = 1
        if tensor.dim() > 2:
            receptive_field_size = tensor[0][0].numel() # which should be kernel_size[0]*kernel_size[1]*kernel_size[2]
        fan_in = num_input_fmaps * receptive_field_size
        fan_out = num_output_fmaps * receptive_field_size
```

I've done a quick test (train 200 epochs once) on the mini-kinetics dataset and by fixing the weights initialization the accuracy seems to improve. 

Let me know if it makes sense.

**Comment by kenshohara** [comment]
It seemed that you are right.
I will try to fix the problem and confirm whether the accuracies improve or not.
Thank you for your information!

**Comment by kenshohara** [comment]
I conducted the experiment that uses ResNet-18 with the fixed initialization
but I could not confirm significant performance difference.
Anyway, the initialization should be fixed,
and I fixed it.
